### ISSUE TITLE
3 tests fail on FreeBSD

### ISSUE THREAD

**Issue by yurivict** [pika-org/pika#463]
```
          Start  468: tests.unit.modules.assertion.assert_fail
 468/1208 Test  #468: tests.unit.modules.assertion.assert_fail ..................................................................***Failed    0.01 sec
No tests run. Did you forget to add tests?
          Start  574: tests.unit.modules.debugging.print
 574/1208 Test  #574: tests.unit.modules.debugging.print ........................................................................***Exception: SegFault  0.01 sec
          Start  954: tests.unit.modules.program_options.parsers
 954/1208 Test  #954: tests.unit.modules.program_options.parsers ................................................................***Exception: SegFault  0.01 sec
```

Version: 0.8.0
OS: FreeBSD 13.1 STABLE

**Comment by msimberg** [comment]
Could I ask you to rerun those tests with `ctest --verbose` and report back?

My guess is that the first test failure is fixed by #395 (specifically this https://github.com/pika-org/pika/pull/395/commits/684fb53fff886942fa07ddf14e3cfc2acd9069d6#diff-feda20a74f31a090b8742624cbf7598986d2bcd2d74f84ea8e9b7b519a9394a9, it's a false positive in release mode). The others may be legitimate issues but it's hard to say without the test output.

**Comment by msimberg** [comment]
I'm going to close this in favour of #895. It looks like this is a subset of the reported failures in the other issue.
